### PR TITLE
[`flake8-pyi`] Implement PYI041

### PR DESCRIPTION
**Pull request by density**
## Summary

Implements PYI041 from flake8-pyi. See [original code](https://github.com/PyCQA/flake8-pyi/blob/2a86db8271dc500247a8a69419536240334731cf/pyi.py#L1283).

This check only applies to function parameters in order to avoid issues with mypy. See https://github.com/PyCQA/flake8-pyi/issues/299.

ref: #848

## Test Plan

Snapshots, manual runs of flake8.

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+async def f4(**kwargs: int | int | float) -> None:
+    ...  # PYI041
+
+
+class Foo:
+    def good(self, arg: int) -> None:
+        ...
+
+    def bad(self, arg: int | float | complex) -> None:
+        ...  # PYI041
```
Comment (dhruvmanila) [inline crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.py:47]: nit: I don't think these rules are checked on Python (`.py`) files

```suggestion
def f0(arg1: float | int) -> None:
    ...


def f1(arg1: float, *, arg2: float | list[str] | type[bool] | complex) -> None:
    ...


def f2(arg1: int, /, arg2: int | int | float) -> None:
    ...


def f3(arg1: int, *args: Union[int | int | float]) -> None:
    ...


async def f4(**kwargs: int | int | float) -> None:
    ...


class Foo:
    def good(self, arg: int) -> None:
        ...

    def bad(self, arg: int | float | complex) -> None:
        ...
```

File: crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.pyi [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Union,
+)
+
+from typing_extensions import (
+    TypeAlias,
+)
+
+TA0: TypeAlias = int
+TA1: TypeAlias = int | float | bool
+TA2: TypeAlias = Union[int, float, bool]
```
Comment (dhruvmanila) [inline crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.pyi:11]: I think we should document the limitation that this rule won't check for type aliases.

**Review by dhruvmanila — APPROVED** [review]
Looks good!

File: crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.py (hunk 1 above)
Comment (density) [reply crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.py]: Fixed, thanks!

File: crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.pyi (hunk 2 above)
Comment (density) [reply crates/ruff/resources/test/fixtures/flake8_pyi/PYI041.pyi:12]: Added a comment. Thanks!

**Pull request merged**